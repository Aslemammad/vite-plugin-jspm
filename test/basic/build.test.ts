import sirv from "sirv";
import puppeteer from "puppeteer";
import { build, preview, PreviewServer } from "vite";
import { describe, afterAll, beforeAll, expect, test } from "vitest";

const url = "http://localhost:3000";

describe("build", async () => {
  await build({ configFile: "./vite.config.ts" });
  let server: PreviewServer;

  let browser: puppeteer.Browser;
  let page: puppeteer.Page;

  beforeAll(async () => {
    server = await preview({
      configFile: "./vite.config.ts",
      preview: { open: false, port: 3000 },
    });
    server.printUrls();

    browser = await puppeteer.launch();
    page = await browser.newPage();
  });

  test("basic render", async () => {
    await page.goto(url);
    expect(await page.content()).toContain("Hello, world!");
  });

  test("contains importmap", async () => {
    expect(await page.content()).toContain('<script type="importmap">');

    expect(await page.content()).toContain(`{
  "imports": {
    "react": "https://ga.jspm.io/npm:react@17.0.2/index.js",
    "react-dom": "https://ga.jspm.io/npm:react-dom@17.0.2/index.js"
  },
  "scopes": {
    "https://ga.jspm.io/": {
      "object-assign": "https://ga.jspm.io/npm:object-assign@4.1.1/index.js",
      "scheduler": "https://ga.jspm.io/npm:scheduler@0.20.2/index.js"
    }
  }
}`);

  expect(await page.content()).toMatchSnapshot()
  });

  afterAll(async () => {
    await browser.close();
    server.httpServer.close();
  });
});
