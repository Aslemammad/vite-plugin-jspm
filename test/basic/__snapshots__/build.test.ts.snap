// Vitest Snapshot v1

exports[`build > contains importmap 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head><script async=\\"\\" src=\\"https://ga.jspm.io/npm:es-module-shims@1.4.1/dist/es-module-shims.js\\"></script>
<script type=\\"importmap\\">{
  \\"imports\\": {
    \\"react\\": \\"https://ga.jspm.io/npm:react@17.0.2/index.js\\",
    \\"react-dom\\": \\"https://ga.jspm.io/npm:react-dom@17.0.2/index.js\\"
  },
  \\"scopes\\": {
    \\"https://ga.jspm.io/\\": {
      \\"object-assign\\": \\"https://ga.jspm.io/npm:object-assign@4.1.1/index.js\\",
      \\"scheduler\\": \\"https://ga.jspm.io/npm:scheduler@0.20.2/index.js\\"
    }
  }
}</script>
    <meta charset=\\"UTF-8\\">
    <link rel=\\"icon\\" type=\\"image/svg+xml\\" href=\\"/src/favicon.svg\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <title>Vite App</title>
    <script type=\\"module\\" crossorigin=\\"\\" src=\\"/assets/index.0fb49565.js\\"></script>
  </head>
  <body>
    <div id=\\"root\\"><div><h1>Hello, world!</h1></div></div>
    
  

</body></html>"
`;
